src/marqo/tensor_search/utils.py

```diff
@@ -175,20 +176,52 @@ def read_env_vars_and_defaults(var: str) -> Optional[str]:
     """Attempts to read an environment variable.
     If none is found, it will attempt to retrieve it from
     configs.default_env_vars(). If still unsuccessful, None is returned.
+    If it's an empty string, None is returned.
     """
-    try:
-        var = os.environ[var]
-        if var is not None and len(var) == 0:
+
+    def none_if_empty(value: Optional[str]) -> Optional[str]:
```

Comment:
nit: function in a function is often an anti-pattern.

Reply:
Agreed